### ISSUE TITLE
Support for Lumen

### ISSUE THREAD

**Issue by DiederikvandenB** [aacotroneo/laravel-saml2#40]
Quick question, are you planning on supporting Lumen anytime in the near future? There are only a few small issues to be ironed out I believe.

**Comment by aacotroneo** [comment]
I dont have the time now to do that, but if anyone wants to make a fork/branch/PR is welcome

**Comment by ChinmayPatel** [comment]
Do you have a list of tasks to be done to support Lumen. I may be able to take a look at this in the coming weeks.

**Comment by aacotroneo** [comment]
Sorry @ChinmayPatel I've been away of php for a while.. I've never used Lumen at all. I imagine the hooks should be similar.. this library just grabs configuration and pass control to one-long at proper times

**Comment by ChinmayPatel** [comment]
Hey @aacotroneo. I'll give it a try in the coming weeks if we decide to go with SAML. Thanks for the prompt response.

**Comment by denizdogan** [comment]
I'm also very interested in using this for a Lumen project, so if anyone's figured it out, I'd gladly hear about your experiences using it

**Comment by durimjusaj** [comment]
Hey there, anybody made this to support lumen or does anybody have any list what we need to do to get this working on lumen ?

**Comment by danmichaelo** [comment]
Of couse it never hurts with Lumen support, but imho Lumen is best for simple, session-less services, and if I needed SAML I would go for Laravel instead. From Matt Stauffer: "Lumen is built for microservices, not so much for user-facing applications (although it can be used for anything.) " (https://mattstauffer.co/blog/introducing-lumen-from-laravel)

**Comment by lelandpalmer** [comment]
Has Lumen support been released yet? Or maybe someone made the necessäry changes in order to implement this package on Lumen ? I am really looking forward to implement it in an existing Lumen microservice

**Comment by ibpavlov** [comment]
Hey, I also needed Lumen version and forked this repo to create another -[https://packagist.org/packages/ibpavlov/lumen-saml2](https://packagist.org/packages/ibpavlov/lumen-saml2). I prefer separate repo because of the tag versioning. 

@danmichaelo You are right about the microservices, but there are other good use cases. We use Lumen with a legacy system to enhance it, add new features and clean the code by using Illuminate packages.

**Comment by Rushi1807** [comment]
@ibpavlov  do you have any working sample lumen project for the same if yes please share.Thanks in advance.

**Comment by ibpavlov** [comment]
> @ibpavlov do you have any working sample lumen project for the same if yes please share.Thanks in advance.

No. I use it in a private project. But this ia a good idea. I'll try to create one soon.

**Comment by Rushi1807** [comment]
@ibpavlov  We are unable configure your package, we followed all steps from read me still facing issue in setting up with one login if you have any other related documents with setup please help!

**Comment by ibpavlov** [comment]
Hey. I've used [https://github.com/onelogin/php-saml](https://github.com/onelogin/php-saml) documentation to setup this package. We are using Microsoft Azure AD with Saml 2 in our project.

**Comment by Rushi1807** [comment]
@ibpavlov  we are facing Logger and Session issue in saml2Controller file. please help!Thanks

**Comment by ibpavlov** [comment]
> @ibpavlov we are facing Logger and Session issue in saml2Controller file. please help!Thanks

@Rushi1807 
You can write your own controller and change config for 'saml2_settings.saml2_controller'. Methids `acl` and `sls` just use the logger for debugging purposes. But some session storage is required here to keep the user authenticated.

**Comment by Rushi1807** [comment]
@ibpavlov Can you help me with this error!Thanks
`sprintf(): Too few argumentsArray ( [error] => Array ( [0] => invalid_response ) [last_error_reason] => sprintf(): Too few arguments )`

**Comment by ibpavlov** [comment]
@Rushi1807 You have to check your code for `sprintf` usage and check the arguments. Or you should provide more information.

**Comment by Rushi1807** [comment]
@ibpavlov Are there any extra configurations needed in the same package for Microsoft ADFS implementation. It worked for OneLogin toolkit Thank you

**Comment by ibpavlov** [comment]
@Rushi1807 If it supports Saml2 protocol, then yes. The configurations itself are the same for all different Identity providers (IDP) with SAML2. We have used it with Microsoft Azure AD.

**Comment by lloricode** [comment]
any plan to support for lumen? this is much better to integrate in micro framework since this is not needed to use other web component in laravel

**Comment by arturslogins** [comment]
Any progress with this issue?
